### DISCUSSION TITLE
Reversed shop scenario

### DISCUSSION THREAD

**Discussion by amadeus-torwell** [vendurehq/vendure#1395] [Q&A] [answered]
Hi everyone,

I just started looking into vendure - and I have a design question.

Is it theoretical / architectural possible (or is there even something in code) that would enable vendure to act as a reversed shop? Basically this means I want to send people money (upfront) for sending items to the shop (including shipping etc.).

Thanks 👍 
Amadeus

**Comment by michaelbromley** [comment]
Hi, I've not come across this concept before. Can you go into more detail about the exact work flow you'd like to support? Like a high-level step by step list of actions & transactions taken by the parties involved?

**Comment by amadeus-torwell** [comment]
Sure, instead of selling an item to a customer think of buying an item through the shop from a customer:

1) Shop lists an item "Guitar Gibson" for up to $1.000
2) Customer want's to sell his Gibson guitar in medium condition
3) Shop asks some questions like "Does your guitar have any scratches?", "How old is your guitar?" -> Some algorithm (external service) estimates a price (let's say $700)
4) Customer prints shipping label and sends guitar to the shop owner
5) Money is flowing from the shop to the customer (via PayPal or whatever payment service)

**Comment by michaelbromley** [comment, marked as answer]
Thanks for the explanation. I think it might be possible to support this with Vendure. Here's how it might be workable:

1. A ProductVariant in Vendure would correspond to a "listing".
2. When a customer clicks "bid on this listing" you add that ProductVariant to the Order (`addItemToOrder` mutation)
3. Answers to the questions can be stored in [custom fields on the OrderLine entity](https://www.vendure.io/docs/developer-guide/customizing-models/#configurable-order-products).
4. You'd need a custom PaymentMethodHandler which doesn't actually take any payment.
5. You'd use a [custom order process](https://www.vendure.io/docs/developer-guide/customizing-the-order-process/) to model any steps between "checkout" (customer making bid on listing), and "shipping" (sending payment to customer).
